### ISSUE TITLE
New Rule: component-destructure-props

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint-react#27]
Enforces destructuring of `this.props` fields inside rendering methods in a component class to encourage succinctness.

good:

``` ts
public render() {
  const { foo, bar } = this.props;
  return <SomeComponent foo={bar} bar={foo} />;
}
```

bad:

``` ts
public render() {
  return <SomeComponent foo={this.props.bar} bar={this.props.foo} />;
}
```

Destructuring also allows you to address conflicts with other identifiers in scope:

``` ts
const foo: string;

public render() {
  const { foo: awesomeFoo, bar } = this.props;
  return <SomeComponent foo={bar} bar={awesomeFoo} />;
}
```

**Comment by adidahiya** [comment]
Should be covered by https://github.com/yannickcr/eslint-plugin-react/blob/97a9f397d2a8a3ce3b2af893bdbb86bb2c1d4480/docs/rules/destructuring-assignment.md after migration to eslint
